### PR TITLE
add a UDP interface for PPRZLINK python library

### PR DESCRIPTION
**Pull request by gautierhattenberger**
similar to serial and ivy interface

### CODE REVIEW

File: lib/v1.0/python/pprzlink/udp.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import threading
+import socket
+
+# used to run the test from this directory
+# any better implementation ?
+import os
+import sys
+PACKAGE_PARENT = '..'
+SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.join(os.getcwd(), os.path.expanduser(__file__))))
+sys.path.append(os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
```
Comment (gautierhattenberger) [inline lib/v1.0/python/pprzlink/udp.py]: @flixr this is the only way I found to be able to run the test code. Is there an other way or is this correct ?

Comment (flixr) [inline lib/v1.0/python/pprzlink/udp.py]: use a relative import like in https://github.com/paparazzi/pprzlink/blob/master/lib/v1.0/python/pprzlink/serial.py#L6

Comment (gautierhattenberger) [inline lib/v1.0/python/pprzlink/udp.py]: That is what I did in the beginning, but then if I try to run the test script with `python serial.py` for instance, I get:
```
Traceback (most recent call last):
  File "serial.py", line 4, in <module>
    import serial
  File "/home/gautier/dev/pprzlink/lib/v1.0/python/pprzlink/serial.py", line 6, in <module>
    from .message import PprzMessage
ValueError: Attempted relative import in non-package
```
(One of) the workaround I found on Internet is the one I used in `udp.py`. But maybe I missed something.

Comment (flixr) [inline lib/v1.0/python/pprzlink/udp.py]: Yeah, it won't work if you don't add .../pprzlink/lib/v1.0/python to you PYTHONPATH before.
But the relative import should work as expected if you do that, e.g. if you want to use the UDP interface in other code..
So for just running the tests, I would say add a README.md to the pprzlink dir that explains that you need to set your PYTHONPATH if you want to run the tests...

Comment (flixr) [inline lib/v1.0/python/pprzlink/udp.py]: You can run the tests with `python -m pprzlink.serial` or `python -m pprzlink.udp` when you have set your PYTHONPATH

**Comment by gautierhattenberger** [comment]
should be fine now

**Pull request merged**